lib/class/search.class.php

```diff
@@ -1386,6 +1434,157 @@ private function video_to_sql()
     }
 
     /**
+     *  movie_to_sql
+     *
+     * Handles the generation of the SQL for video searches.
+     */
+    private function movie_to_sql()
+    {
+        $sql_logic_operator = $this->logic_operator;
+
+        $where = array();
+        $table = array();
+        $join = array();
+        $group = array();
+        $having = array();
+
+        foreach ($this->rules as $rule) {
+            $type = $this->name_to_basetype($rule[0]);
+            $operator = array();
+            foreach ($this->basetypes[$type] as $op) {
+                if ($op['name'] == $rule[1]) {
+                    $operator = $op;
+                    break;
+                }
+            }
+            $input = $this->_mangle_data($rule[2], $type, $operator);
+            $sql_match_operator = $operator['sql'];
+
+            switch ($rule[0]) {
+                case 'title':
+                    $where[] = "`video`.`title` $sql_match_operator '$input'";
+                  break;
+                case 'original_name':
+                    $where[] = "`movie`.`original_name` $sql_match_operator '$input'";
+                break;
+                case 'year':
+                    $where[] = "`movie`.`year` $sql_match_operator '$input'";
+                break;
+                case 'certification':
+                    $where[] = "`movie`.`content_rating` $sql_match_operator '$input'";
+                 break;
+                default:
+                    // WE WILLNA BE FOOLED AGAIN!
+            } // switch on ruletype
+        } // foreach rule
+
+        $join['catalog'] = AmpConfig::get('catalog_disable');
+        $join['movie'] = true;
+
+        $where_sql = implode(" $sql_logic_operator ", $where);
+
+        if ($join['catalog']) {
+            $table['catalog'] = "LEFT JOIN `catalog` AS `catalog_se` ON `catalog_se`.`id`=`video`.`catalog`";
+            $where_sql .= " AND `catalog_se`.`enabled` = '1'";
+        }
+
+        if ($join['movie']) {
+            $table['movie'] = "LEFT JOIN `movie` ON `video`.`id` = `movie`.`id`";
+        }
+
+        if ($join['video']) {
+            $table['video'] = "LEFT JOIN `movie` ON `video`.`id` = `movie`.`id`";
+        }
+
+
+        $table_sql = implode(' ', $table);
+        $group_sql = implode(', ', $group);
+        $having_sql = implode(" $sql_logic_operator ", $having);
+
+        return array(
+            'base' => 'SELECT DISTINCT(`video`.`id`) FROM `video`',
+            'join' => $join,
+            'where' => $where,
+            'where_sql' => $where_sql,
+            'table' => $table,
+            'table_sql' => $table_sql,
+            'group_sql' => $group_sql,
+            'having_sql' => $having_sql
+        );
+    }
+
+    /**
+     *  tvshow_to_sql
+     *
+     * Handles the generation of the SQL for TV show searches.
+     */
+    private function tvshow_to_sql()
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method tvshow_to_sql is not named in camelCase.](https://www.codacy.com/app/paul/ampache/file/616134479/issues/source?bid=7631&fileBranchId=2290754#l1521)